test/debug_info/cmn_blk_dbg_info.f90

```diff
@@ -0,0 +1,25 @@
+!RUN: %flang -gdwarf-4 -S -emit-llvm %s -o - | FileCheck %s
+
+!CHECK: @blk_ = global {{.*}} align 64, !dbg !{{[0-9]+}}, !dbg !{{[0-9]+}}, !dbg !{{[0-9]+}}, !dbg !{{[0-9]+}}, !dbg !{{[0-9]+}}, !dbg !{{[0-9]+}}
```

Comment:
I ran this test case, and have two questions.

1. There are actually 9 `!dbg` metadata nodes associated with `@blk_`. The FileCheck pattern here matches only six. Moreover, you are not checking that the metadata refer to the declarations of `b3i` and `b3j` on three different lines, which I feel is the purpose of this test.
2. By debugging `llassem.cpp`, I found that this test does not actually exercise the newly added code following `if (cmn_blk_ir) {` (line 1738), even though the output IR does contain the additional `!dbg` metadata nodes as expected. Isn't this test supposed to provide coverage for the newly added code?

Reply:
Earlier, the intention was to check whether the global "@blk" common block variable had at least 6 reference one for ever b3i and b3j in 3 different scopes. Before the fix, only 2 metadata nodes were associated with @blk even though it were present in the "ll" file.
Now changed the CHECKs to verify that "@blk" variable associated metadata nodes are comes from every scope.
Please let me know if you feel too many CHECK lines. I restrict to one variable for each scope.

Reply:
That looks much better. Thanks!